### PR TITLE
Even slimmer single thread start pr

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: sn_interface/src/network_knowledge/mod.rs [hunk 1]
```diff
@@ -580,6 +577,9 @@ impl NetworkKnowledge {
             }
         }
 
+        // drop the borrow
+        drop(chain);
```
Comment (maqi) [inline sn_interface/src/network_knowledge/mod.rs:581]: why for the `self.signed_sap`, there is no explicit `drop` after the `borrow`, but here need a drop for the `self.chain` ?
note the correspondent calling place is a `clone` call: `let chain = self.chain.borrow().clone();`

File: sn_interface/src/network_knowledge/mod.rs [hunk 2]
```diff
@@ -593,15 +593,18 @@ impl NetworkKnowledge {
             node_name,
             node_state.state()
         );
+
+        let the_chain = self.chain.borrow();
```
Comment (maqi) [inline sn_interface/src/network_knowledge/mod.rs:597]: what if call with `self.section_chain()` to save the `drop` ?

File: sn_interface/src/network_knowledge/mod.rs (hunk 1 above)
Comment (joshuef) [reply sn_interface/src/network_knowledge/mod.rs:581]: cos we can't hold `chain` across the next `await` point, so we want to drop it.

File: sn_node/examples/routing_stress.rs [hunk 3]
```diff
@@ -227,7 +227,7 @@ impl Network {
         };
 
         // TODO: Fix this: future returned by `add_node` is not `Send`
-        //let _handle = task::spawn(add_node(id, config, event_tx));
+        //let _handle = task::spawn_local(add_node(id, config, event_tx));
```
Comment (oetyng) [inline sn_node/examples/routing_stress.rs:230]: Maybe this todo can be fixed now, by uncommenting line 230 and removing line 229 and 231?

Comment (oetyng) [reply sn_node/examples/routing_stress.rs:230]: Just noticed its `routing_stress.rs`. Does this stress test even work anymore?

Comment (joshuef) [reply sn_node/examples/routing_stress.rs:230]: Okay, I have a fix that gets it copmiling. I think we can address its relevance / CI worthiness apart from this PR.

File: sn_interface/src/network_knowledge/mod.rs (hunk 1 above)
Comment (joshuef) [reply sn_interface/src/network_knowledge/mod.rs:581]: ah so it is now!

**Review by maqi — APPROVED** [review]

**Review by davidrusu — APPROVED** [review]

**Comment by davidrusu** [comment]
bors retry

File: sn_node/examples/routing_stress.rs (hunk 3 above)
Comment (oetyng) [reply sn_node/examples/routing_stress.rs]: > Maybe this todo can be fixed now, by uncommenting line 230 and removing line 229 and 231?

And what about this thing?

**Pull request merged**